### ISSUE TITLE
Filter criteria not getting cleared properly when no results are yielded

### ISSUE THREAD

**Issue by sguptatw** [getgauge/html-report#113]
**Expected behavior**
If a filter criteria does not yield any results. On clearing the criteria, the html-report should show the default specs and scenarios.

**Actual behavior**
If a filter criteria does not yield any results. On clearing the criteria, the html-report still shows no specs and scenarios.

**Steps to replicate**
1. On gauge-tests html-report, apply filter `def`.
2. It gives no spec and scenarios.
3. Now clear the filter, it does not show the default specs and scenarios.
4. Clicking on passed and failed statistics also does not change the behavior.

**Version**
```
Gauge version: 0.6.3.nightly-2016-11-27

Plugins
-------
csharp (0.9.2.nightly-2016-09-14)
html-report (2.2.0.nightly-2016-11-27)
java (0.5.1.nightly-2016-11-21)
js (1.3.0)
ruby (0.3.2.nightly-2016-08-22)
spectacle (0.0.1)
xml-report (0.1.4.nightly-2016-10-27)
```

**Comment by apoorvam** [comment]
Fix should be available in nightly >= 29.11.2016

**Comment by sguptatw** [comment]
failing in the following scenario
- retain the criteria not giving results and click on statistics - passed,failed, skipped.

The scenario list is empty but the scenario details is still showing some info.

**Comment by apoorvam** [comment]
Fix should be available in nightly >= 6.12.2016
